### PR TITLE
VCARDS don't hold binary data

### LINKED ISSUE #2258: Sabre\VObject\ParseException: Invalid Mimedir file. Line starting at 11 did not follow iCalendar/vCard conventions
After uploading an avatar: 

```
Sabre\VObject\ParseException: Invalid Mimedir file. Line starting at 11 did not follow iCalendar/vCard conventions
/media/psf/nextcloud/3rdparty/sabre/vobject/lib/Parser/MimeDir.php - line 234: Sabre\VObject\Parser\MimeDir->readProperty('\xFF\xDB\x00C\x00\x08\x06\x06\x07\x06\x05\x08\x07\x07\x07...')
/media/psf/nextcloud/3rdparty/sabre/vobject/lib/Parser/MimeDir.php - line 181: Sabre\VObject\Parser\MimeDir->parseLine('\xFF\xDB\x00C\x00\x08\x06\x06\x07\x06\x05\x08\x07\x07\x07...')
/media/psf/nextcloud/3rdparty/sabre/vobject/lib/Parser/MimeDir.php - line 89: Sabre\VObject\Parser\MimeDir->parseDocument()
/media/psf/nextcloud/3rdparty/sabre/vobject/lib/Reader.php - line 46: Sabre\VObject\Parser\MimeDir->parse('BEGIN VCARD\r\nVE...', 0)
/media/psf/nextcloud/apps/dav/lib/CardDAV/CardDavBackend.php - line 985: Sabre\VObject\Reader read('BEGIN VCARD\r\nVE...')
/media/psf/nextcloud/apps/dav/lib/CardDAV/CardDavBackend.php - line 944: OCA\DAV\CardDAV\CardDavBackend->readCard('BEGIN VCARD\r\nVE...')
/media/psf/nextcloud/apps/dav/lib/CardDAV/CardDavBackend.php - line 608: OCA\DAV\CardDAV\CardDavBackend->updateProperties('1', 'Database banana...', 'BEGIN VCARD\r\nVE...')
/media/psf/nextcloud/apps/dav/lib/CardDAV/SyncService.php - line 247: OCA\DAV\CardDAV\CardDavBackend->updateCard('1', 'Database banana...', 'BEGIN VCARD\r\nVE...')
/media/psf/nextcloud/apps/dav/lib/HookManager.php - line 120: OCA\DAV\CardDAV\SyncService->updateUser(Object(OC\User\User))
/media/psf/nextcloud/lib/private/legacy/hook.php - line 106: OCA\DAV\HookManager->changeUser(Array)
/media/psf/nextcloud/lib/private/Server.php - line 309: OC_Hook emit('OC_User', 'changeUser', Array)
[internal function] OC\Server->OC\{closure}(Object(OC\User\User), 'avatar', NULL)
/media/psf/nextcloud/lib/private/Hooks/EmitterTrait.php - line 99: call_user_func_array(Object(Closure), Array)
/media/psf/nextcloud/lib/private/Hooks/PublicEmitter.php - line 33: OC\Hooks\BasicEmitter->emit('\\OC\\User', 'changeUser', Array)
/media/psf/nextcloud/lib/private/User/User.php - line 427: OC\Hooks\PublicEmitter->emit('\\OC\\User', 'changeUser', Array)
/media/psf/nextcloud/lib/private/Avatar.php - line 139: OC\User\User->triggerChange('avatar')
/media/psf/nextcloud/core/Controller/AvatarController.php - line 340: OC\Avatar->set('\xFF\xD8\xFF\xE0\x00\x10JFIF\x00\x01\x01\x01\x00...')
[internal function] OC\Core\Controller\AvatarController->postCroppedAvatar(Array)
/media/psf/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 160: call_user_func_array(Array, Array)
/media/psf/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 90: OC\AppFramework\Http\Dispatcher->executeController(Object(OC\Core\Controller\AvatarController), 'postCroppedAvat...')
/media/psf/nextcloud/lib/private/AppFramework/App.php - line 114: OC\AppFramework\Http\Dispatcher->dispatch(Object(OC\Core\Controller\AvatarController), 'postCroppedAvat...')
/media/psf/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47: OC\AppFramework\App main('OC\\Core\\Control...', 'postCroppedAvat...', Object(OC\AppFramework\DependencyInjection\DIContainer), Array)
[internal function] OC\AppFramework\Routing\RouteActionHandler->__invoke(Array)
/media/psf/nextcloud/lib/private/Route/Router.php - line 299: call_user_func(Object(OC\AppFramework\Routing\RouteActionHandler), Array)
/media/psf/nextcloud/lib/base.php - line 1005: OC\Route\Router->match('/avatar/cropped')
/media/psf/nextcloud/index.php - line 40: OC handleRequest()
{main}
```

cc @icewind1991

### PR DESCRIPTION
**Pull request by rullzer**
Should fix #2258

We need to base64 encode the avatar and not just dump binary data in there.

TODO:
- [x] Add tests

### CODE REVIEW

File: .htaccess [hunk 1]
```diff
@@ -77,3 +77,7 @@ Options -Indexes
 <IfModule pagespeed_module>
   ModPagespeed Off
 </IfModule>
+#### DO NOT CHANGE ANYTHING ABOVE THIS LINE ####
+
+ErrorDocument 403 /core/templates/403.php
+ErrorDocument 404 /core/templates/404.php
```
Comment (LukasReschke) [inline .htaccess]: 😉

**Review by LukasReschke — DISMISSED** [review]

**Comment by nickvergessen** [comment]
Works, but should have an update step, or is this only a problem for avatars that where added on master after the accounts stuff was merged

**Comment by rullzer** [comment]
It seems that data currently in the DB is correct. Maybe there was some sabre magic that converted it to base64 before

**Comment by schiessle** [comment]
> It seems that data currently in the DB is correct. Maybe there was some sabre magic that converted it to base64 before

yes, strange. First I thought that I introduced it here https://github.com/nextcloud/server/commit/8f33d9d1c07c164b18be9d36e9e9f79b5bdfa8be but also before there was no explicit base64 encoding.

If your assumption is correct that previous SabreDAV version did this internally, then we should double check if reading this cards still work or if we now also need to decode the photo explicitly. Did you tried it?

**Comment by rullzer** [comment]
I tried on my own NC10 instance and there it is just stored as base64 encoded data, but will try more tests

**Comment by schiessle** [comment]
> I tried on my own NC10 instance and there it is just stored as base64 encoded data, but will try more tests

yes, also try to read them... Are they displayed again correctly in Nextcloud 11, base64 encoded, e.g. in the contacts app?

**Comment by rullzer** [comment]
Ok please review. I moved the scheme to proper vcard v4.
Clicking around and all seemed to work well here.

Please test.

**Review by LukasReschke — APPROVED** [review]
LGTM

**Comment by LukasReschke** [comment]
LGTM

**Comment by MorrisJobke** [comment]
Tested and works 👍

**Pull request merged**